### DISCUSSION TITLE
WSA Android 13 beta test release link had been captured

### DISCUSSION THREAD

**Discussion by someone5678** [LSPosed/MagiskOnWSALocal#298] [General]
File:
[MicrosoftCorporationII.WindowsSubsystemForAndroid_2211.40000.10.0_neutral___8wekyb3d8bbwe.Msixbundle](http://tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/6b10fc29-a607-47c1-98b6-8808d7839656?P1=1671129420&P2=404&P3=2&P4=C7vnXvpHM0MsIW5fPhBhJPHPPd69kzCViReYX0WMZftf1KL2VFah6COI0tf4VAvwanD5mRaQz7tRKq9duFnkMg%3d%3d)
(Right click -> Copy link address)

Used Software:
[Fiddler Classic](https://www.telerik.com/download/fiddler)

Following this guide:
[Windows OS Hub: How to Download APPX File from Microsoft Store for Offline Installation?](http://woshub.com/how-to-download-appx-installation-file-for-any-windows-store-app/)

Search tlu. (part of link) instead of appx

**Comment by williamvyong** [comment]
do u make it download ?
cause i tried every method i can 
but found nothing

**Comment by Howard20181** [comment]
https://github.com/Howard20181/WSA-bundle/releases/tag/v2211.40000.10.0

**Comment by someone5678** [comment]
Turns out, We can get the link by replacing <device></device> flag with <user></user> sub flag in
`<TicketType Name="MSA" Version="1.0" Policy="MBI_SSL"></TicketType>`
flag in the file MagiskOnWSALocal/xml/{FE3FileUrl.xml, WUIDRequest.xml}

e.g.

```
<TicketType Name="MSA" Version="1.0" Policy="MBI_SSL">
<user>your_user_code
</user>
</TicketType>
```

So, I think sharing your user code is not a good idea... for obvious reason

This, also can be inspectable by Fiddler by following above post
And then searching `ClientWebService/client.asmx`
click the TextView of top right
And then search `<user>`

**Reply by Howard20181** [reply]
What is the usercode? Microsoft account ID?

**Reply by Howard20181** [reply]
Need to capture the store's network traffic to get the user ID, can't be generated automatically
Customization is now supported, please check if it works properly

**Reply by someone5678** [reply]
> What is the usercode? Microsoft account ID?

I can't explain well actually... because I don't know that much
They seems like make a encrypted key(base64?) based on Microsoft account and granted wsa beta download per user, not per ring(WIF WIS retail etc).
Anyways, you need Fiddler or similar software for find that.

If you can't get such stuff while download WSA using Fiddler, just download any other Microsoft Store released app(e.g. Windows Scan) while Fiddler is on. There, you can find the session which it's url is `ClientWebService/client.asmx`.

**Reply by someone5678** [reply]
FYI, This is what will looks like
![Screenshot 2022-12-18 203438](https://user-images.githubusercontent.com/59456192/208296155-1b971ed0-d569-4ed2-a531-d1151a4169f4.png)

**Reply by s1204IT** [reply]
Memo:
secured . s:Envelope . s:Header . o:Security . wuws:WindowsUpdateTicketsToken . TicketType  .User

**Reply by dongle-the-gadget** [reply]
> What is the usercode? Microsoft account ID?

The usercode here is either an XSTS token or MSA token format.

You can reference https://github.com/gus33000/UUPMediaCreator/blob/master/src/WindowsUpdateLib.Shared/MBIHelper.cs for the MSA token format (this one works in the command line and requires the MSA email and password - or app password for 2FA accounts) and https://github.com/OpenXbox/xbox-webapi-python for XSTS (this one requires Web OAuth).

**Comment by GuyInCanada** [comment]
update to 2211.40000.11 (link expires 2023-01-08 5pm EST)
http://tlu.dl.delivery.mp.microsoft.com/filestreamingservice/files/6f031e6b-f66d-4dee-8fe4-7f6f9a85cdc1?P1=1673157595&P2=404&P3=2&P4=JHAb%2f5fXUsr71szaxWfJo9lmcg337ag8sTl1KS6zHcwp5NIlg6TJPRSZd%2bWfyca8vqoiDvK96M2ShxYy6QgGEA%3d%3d

**Reply by s1204IT** [reply]
I mirrored after downloaded from your link.
thanks.
https://www.mediafire.com/file/k55muu2lxzgu25e/MicrosoftCorporationII.WindowsSubsystemForAndroid_2211.40000.11.0_neutral_8wekyb3d8bbwe.Msixbundle/file

**Reply by DeepBlueBubble** [reply]
thank you for the sharing!
